### PR TITLE
feat(js-connectors): enable metrics and query tracing

### PR DESCRIPTION
**Pull request by jkomyno**
This PR enables:
- metrics on napi.rs-based JS Connectors' queries:
  - This currently re-uses whatever Quaint used. We're going to have to introduce some conditional logic in there for Phase 2. (E.g., `std::time::Instant` will likely not work on `wasm32-unknown-unknown`).
- query tracing (`prisma.$on('query', () => { ... })`) on napi.rs-based JS Connectors' queries

This PR closes https://github.com/prisma/team-orm/issues/330 and https://github.com/prisma/team-orm/issues/335.

### CODE REVIEW

File: quaint/src/connector.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 //! querying interface.
 
 mod connection_info;
-pub(crate) mod metrics;
+pub mod metrics;
```
Comment (tomhoule) [inline quaint/src/connector.rs:13]: nitpick: we try to follow a convention for grouping imports and exports in the codebase. This should be grouped with the public exports. In general, it looks like this:

```rust
pub mod a;
pub mod b;

mod c;
mod d;

pub use c::Something;

use d::SomethingElse;

```

File: quaint/src/connector/metrics.rs [hunk 2]
```diff
@@ -3,12 +3,7 @@ use tracing::{info_span, Instrument};
 use crate::ast::{Params, Value};
 use std::{future::Future, time::Instant};
 
-pub(crate) async fn query<'a, F, T, U>(
-    tag: &'static str,
-    query: &'a str,
-    params: &'a [Value<'_>],
-    f: F,
-) -> crate::Result<T>
+pub async fn query<'a, F, T, U>(tag: &'static str, query: &'a str, params: &'a [Value<'_>], f: F) -> crate::Result<T>
```
Comment (tomhoule) [inline quaint/src/connector/metrics.rs:6]: I'm not a fan of having this public, but it's the right call, this is the smallest change that works and we're past trying to keep the quaint public API clean.

**Review by tomhoule — APPROVED** [review]
There's a few red things in CI I haven't looked at to fix — approved on the content :+1:

**Comment by jkomyno** [comment]
> There's a few red things in CI I haven't looked at to fix — approved on the content 👍

I'm merging, as I've applied the "nit" changes and the current clippy failures are not due to this PR, as reported on Slack.

**Review by miguelff — COMMENTED** [review]
:100:

**Pull request merged**